runtime/sema/checker.go

```diff
@@ -283,13 +267,18 @@ func NewChecker(program *ast.Program, location common.Location, options ...Optio
 	return checker, nil
 }
 
-// SetAllCheckers sets the given map of checkers as the map of all checkers.
-//
-func (checker *Checker) SetAllCheckers(allCheckers map[common.LocationID]*Checker) {
-	checker.AllCheckers = allCheckers
-
-	// Register self
-	checker.AllCheckers[checker.Location.ID()] = checker
+func (checker *Checker) SubChecker(program *ast.Program, location common.Location) (*Checker, error) {
```

Comment:
Is this function only used for tests?

Reply:
It could be, I'm not sure. It's pretty much a convenience function to easily create a checker for a program which has the same options as the given checker. 

I guess the naming could be improved, though can't think of a better name.